### PR TITLE
#19453 deleting userproxy and the struts classes that had to do with …

### LINKED ISSUE #19453: Remove UserProxy from Hibernate
We need to remove the legacy code used in the UserProxy from hibernate and all the classes that use it.

### PR DESCRIPTION
**Pull request by wezell**
Creating a PR to see test results.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/beans/UserProxy.java [hunk 1]
```diff
@@ -1,710 +1,71 @@
...
-	}
-	
-	private String challengeQuestionId;
-	private String challengeQuestionAnswer;
-	
-	public boolean isNoclicktracking() {
-		return noclicktracking;
-	}
+    @Override
+    public void setOwner(String owner) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/beans/UserProxy.java:48]: Codacy found an issue: [Document empty method body](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/CMSUsersWebAPI.java [hunk 2]
```diff
@@ -1,7 +1,16 @@
 package com.dotcms.rendering.velocity.viewtools;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/CMSUsersWebAPI.java:5]: Codacy found an issue: [Avoid unused imports such as 'java.util.HashMap'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotmarketing/business/UserProxyAPIImpl.java [hunk 3]
```diff
@@ -17,62 +14,40 @@
...
-		UserProxy up =  userProxyFactory.getUserProxyByLongLiveCookie(dotCMSID);
-		if(permissionAPI.doesUserHavePermission(up, PermissionAPI.PERMISSION_READ, user, respectFrontEndRoles)){
-			return up;
-		}else{
-			throw new DotSecurityException("User doesn't have permissions to retrieve UserProxy");
-		}
-	}
+    @WrapInTransaction
+    public void saveUserProxy(UserProxy userProxy, User user, boolean respectFrontEndRoles)
+                    throws DotRuntimeException, DotDataException, DotSecurityException {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/UserProxyAPIImpl.java:48]: Codacy found an issue: [Document empty method body](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotmarketing/beans/UserProxy.java [hunk 4]
```diff
@@ -1,710 +1,71 @@
 package com.dotmarketing.beans;
 
-import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
-
 import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Layout;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/beans/UserProxy.java:6]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.business.Layout'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java [hunk 5]
```diff
@@ -254,33 +255,47 @@ public boolean doesUserHavePermission(Permissionable permissionable, int permiss
...
+	    return Try.of(()-> APILocator.getLayoutAPI().doesUserHaveAccessToPortlet("user", userIn)).getOrElse(false);
+	}
+	
+	
+	
 	@CloseDBIfOpened
 	@Override
 	public boolean doesUserHavePermission(final Permissionable permissionable, int permissionType, final User userIn, final boolean respectFrontendRoles) throws DotDataException {
 	    
+	    User user = (userIn==null || userIn.getUserId()==null) ? APILocator.getUserAPI().getAnonymousUser() : userIn;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java:272]: Codacy found an issue: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/CMSUsersWebAPI.java [hunk 6]
```diff
@@ -1,7 +1,16 @@
 package com.dotcms.rendering.velocity.viewtools;
 
+import java.io.IOException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/CMSUsersWebAPI.java:3]: Codacy found an issue: [Avoid unused imports such as 'java.io.IOException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotmarketing/business/UserProxyAPIImpl.java [hunk 7]
```diff
@@ -17,62 +14,40 @@
...
+    }
+
+    @CloseDBIfOpened
+    public UserProxy getUserProxy(String userId, User user, boolean respectFrontEndRoles)
+                    throws DotRuntimeException, DotSecurityException, DotDataException {
+        if (userId == null) {
+            return null;
+        }
+
+        UserProxy up = new UserProxy(userId);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/UserProxyAPIImpl.java:30]: Codacy found an issue: [Avoid variables with short names like up](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/CMSUsersWebAPI.java [hunk 8]
```diff
@@ -1,7 +1,16 @@
 package com.dotcms.rendering.velocity.viewtools;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/CMSUsersWebAPI.java:8]: Codacy found an issue: [Avoid unused imports such as 'javax.servlet.http.HttpServletResponse'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6346595)

**Review by nollymar — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Pull request merged**